packages/backend/src/open_payments/auth/middleware.test.ts

```diff
@@ -223,17 +223,95 @@ describe('Auth Middleware', (): void => {
       const next = jest.fn().mockImplementation(async () => {
         await expect(
           GrantReference.query().findById(grant.grant)
-        ).resolves.toEqual({
-          id: grant.grant,
-          clientId: grant.clientId
-        })
+        ).resolves.toBeUndefined()
       })
       await expect(middleware(ctx, next)).resolves.toBeUndefined()
       expect(next).toHaveBeenCalled()
       expect(ctx.grant).toEqual(grant)
       scope.isDone()
     }
   )
+  const table: {
+    type: AccessType
+    action: AccessAction
+    description: string
+  }[] = []
+  Object.values(AccessType).forEach((type) => {
+    Object.values(AccessAction).forEach((action) => {
+      // The complete path is not valid for outgoing payments
+      // Create and read are the only valid paths for quotes
+      if (
+        !(
+          type === AccessType.OutgoingPayment && action == AccessAction.Complete
+        ) &&
+        type !== AccessType.Quote
+      ) {
+        table.push({
+          type: type,
+          action: action,
+          description:
+            action === AccessAction.Create
+              ? 'stores grant details'
+              : 'does not store grant details'
+        })
+      } else if (
+        type === AccessType.Quote &&
+        (action === AccessAction.Create || action === AccessAction.Read)
+      ) {
+        table.push({
+          type: type,
+          action: action,
+          description:
+            action === AccessAction.Create
+              ? 'stores grant details'
+              : 'does not store grant details'
+        })
+      }
+    })
+  })
+  test.each(table)(
+    '$description for $type on the $action path',
```

Comment:
Somewhat more concise.
You're right to call it fragile (until `open-payments` library potentially exposes the type/action pairings).
`read-all` `quotes` is actually in the latest spec but not the version currently used in `backend`

```suggestion
  const types = {
    [AccessType.IncomingPayment]: Object.values(AccessAction),
    [AccessType.OutgoingPayment]: Object.values(AccessAction).filter(
      (action) => action !== AccessAction.Complete
    ),
    [AccessType.Quote]: [
      AccessAction.Create,
      AccessAction.Read
      // TODO
      // AccessAction.ReadAll
    ]
  }

  Object.values(AccessType).forEach((type) => {
    for (const action of types[type]) {
      const description =
        action === AccessAction.Create
          ? 'stores grant details'
          : 'does not store grant details'

      test(`${description} for ${type} on the ${action} path`, async (): Promise<void> => {
```